modules/core/src/weaver/suites.scala

```diff
@@ -58,10 +58,9 @@ abstract class RunnableSuite[F[_]] extends EffectSuite[F] {
     effectCompat.sync(run(args)(outcome => effectCompat.effect.delay(report(outcome))))
 }
 
-abstract class MutableFSuite[F[_]] extends RunnableSuite[F]  {
+abstract class MutableBaseFSuite[F[_]] extends RunnableSuite[F]  {
```

Comment:
Okay, so I think you're on the right track, however what I'd like to see is this : 

```scala
trait MutableBaseFSuite[F[_]] {
    type Res 
    // PerTestRes would be replace Res in the "test" methods 
    type PerTestRes 

    def sharedResource: Resource[F, Res] 
    def perTestResource(res: Res) : Resource[F, PerTestRes] 
}
```

and then MutableFSuite would become :  

```scala
trait MutableFSuite[F[_]] extends MutableBaseFSuite[F] {
   type PerTestRes = Res
   final def perTestResource(res: Res) : Resource[F, Res] = Resource.pure(res)  
}
```

The reason for doing it this way is that : 
1. it works for your usecase 
2. people can still share resources a little bit more granularly than your current version, which is "shared XOR per test", whereas this version would be "shared OR per test" 
3. it works on the javascript backend where cross-suite sharing is not possible, but per suite sharing is possible
4. You can still have your `MutableForEachSuite` in userland : 

```scala
trait MutableForEachSuite[F[_]] extends MutableBaseFSuite[F] {
   type Res = Unit 
   def uniqueResource : Resource[F, PerTestRes]
   final def perTestResource(res: Unit) : Resource[F, PerTestRes] = uniqueResource
}  
```

Reply:
@amumurst, just to touch base : what do you think about the suggestion above ?